### PR TITLE
Fix for duplicate input packages

### LINKED ISSUE #160: Empty Response when user input has duplicate packages
When a request is sent to PGM with duplicate packages, the response received is empty. For example:

Request: 
```json
[
    {
        "ecosystem": "maven",
        "comp_package_count_threshold": 5,
        "alt_package_count_threshold": 2,
        "outlier_probability_threshold": 0.88,
        "unknown_packages_ratio_threshold": 0.3,
        "package_list": [
          "io.vertx:vertx-core",
          "io.vertx:vertx-core",
          "io.vertx:vertx-web"
       ]
    }
]		
```

Response:
```json
[
    {
        "alternate_packages": {},
        "companion_packages": [],
        "ecosystem": "maven",
        "missing_packages": [],
        "outlier_package_list": [],
        "package_to_topic_dict": {},
        "user_persona": "1"
    }
]
```

### PR DESCRIPTION
**Pull request by abs51295**
Resolves #160 .

### CODE REVIEW

**Comment by rootAvish** [comment]
Just one critique here, can you add a unit test for this so I know it's working?

**Comment by abs51295** [comment]
Sure @rootAvish :+1:

**Comment by rootAvish** [comment]
@abs51295 It does not 🙂 , I guess you should convert the set back to a list after you've filtered duplicates.

**Comment by abs51295** [comment]
Yes, need to change the test case to include set and not list.

**Comment by rootAvish** [comment]
Have the changes been made in consumers elsewhere or were they not required?

**Comment by rootAvish** [comment]
Inside the codebase, not the tests.

**Comment by abs51295** [comment]
Changes have been made in the consumers elsewhere. Only need to change the tests to check the same.

File: analytics_platform/kronos/src/recommendation_validator.py [hunk 1]
```diff
@@ -203,20 +203,18 @@ def get_filtered_alternate_list(self, alternate_package, outlier_packages):
...
 
         :return : Filtered input package list."""
 
         filtered_input_list = []
-        if input_list is not None and missing_packages is not None:
-            missing_packages_set = set(missing_packages)
-            filtered_input_list = [
-                package for package in input_list if package not in missing_packages_set]
+        if input_set is not None and missing_packages is not None:
+            filtered_input_list = list(input_set.difference(missing_packages))
```
Comment (sara-02) [inline analytics_platform/kronos/src/recommendation_validator.py]: Are you sure about this? `missing_packages` is a list not set.

Comment (sara-02) [reply analytics_platform/kronos/src/recommendation_validator.py]: Looks like it gets taken care by the `difference` function internally.

Comment (abs51295) [reply analytics_platform/kronos/src/recommendation_validator.py]: Yes @sara-02 you can check the [docs](https://docs.python.org/2/library/sets.html) for the clarification. It clearly says this `Note, the non-operator versions of update(), intersection_update(), difference_update(), and symmetric_difference_update() will accept any iterable as an argument.`

Comment (sara-02) [reply analytics_platform/kronos/src/recommendation_validator.py]: Yes, I tried with the `-` for difference operation, and it was throwing an error. But `difference` function is working fine :)

Comment (abs51295) [reply analytics_platform/kronos/src/recommendation_validator.py]: Thanks :)

**Comment by abs51295** [comment]
@rootAvish I have added the test and they have successfully passed :)

**Pull request merged**